api/src/accounts.rs

```diff
@@ -264,6 +265,33 @@ impl Account {
         })
     }
 
+    pub(crate) fn verify_account_or_object_resource(&self) -> Result<(), BasicErrorWith404> {
+        if self.get_account_resource().is_ok() {
```

Comment:
It doesn't matter because this function only checks if either account or object core exists but semantically, these are not mutually exclusive. An object account can have the account resource too (if create_account is called on it)?